Earthfile

```diff
@@ -0,0 +1,399 @@
+  FROM golang:${GO_VERSION}-alpine3.20
+  RUN apk add --no-cache docker jq
+  COPY +helm-setup/helm /usr/local/bin/helm
+  COPY +kind-setup/kind /usr/local/bin/kind
+  COPY +gotestsum-setup/gotestsum /usr/local/bin/gotestsum
+  COPY +go-build-e2e/e2e .
+  COPY --dir cluster test .
+  WITH DOCKER --load crossplane-e2e/crossplane:latest=+image
+    TRY
+      # TODO(negz:) Set GITHUB_ACTIONS=true and use RUN --raw-output when
+      # https://github.com/earthly/earthly/issues/4143 is fixed.
+      RUN gotestsum --no-color=false --format testname --junitfile e2e-tests.xml --raw-command go tool test2json -t -p E2E ./e2e -test.v ${FLAGS}
+    FINALLY
+      SAVE ARTIFACT --if-exists e2e-tests.xml AS LOCAL _output/tests/e2e-tests.xml
+    END
+  END
+
+# hack builds Crossplane, and deploys it to a kind cluster. It runs in your
+# local environment, not a container. The kind cluster will keep running until
+# you run the unhack target. Run hack again to rebuild Crossplane and restart
+# the kind cluster with the new build.
+hack:
+  # TODO(negz): This could run an interactive shell inside a temporary container
+  # once https://github.com/earthly/earthly/issues/3206 is fixed.
+  LOCALLY
+  WAIT
+    BUILD +unhack
+  END
+  COPY +helm-setup/helm .hack/helm
+  COPY +kind-setup/kind .hack/kind
+  COPY (+helm-build/output --CROSSPLANE_VERSION=v0.0.0-hack) .hack/charts
+  WITH DOCKER --load crossplane-hack/crossplane:hack=+image
+    RUN \
+      .hack/kind create cluster --name crossplane-hack && \
+      .hack/kind load docker-image --name crossplane-hack crossplane-hack/crossplane:hack && \
+      .hack/helm install --create-namespace --namespace crossplane-system crossplane .hack/charts/crossplane-0.0.0-hack.tgz \
+        --set "image.pullPolicy=Never,image.repository=crossplane-hack/crossplane,image.tag=hack" \
+        --set "args={--debug}"
+  END
+  RUN docker image rm crossplane-hack/crossplane:hack
+  RUN rm -rf .hack
+
+# unhack deletes the kind cluster created by the hack target.
+unhack:
+  LOCALLY
+  COPY +kind-setup/kind .hack/kind
+  RUN .hack/kind delete cluster --name crossplane-hack
+  RUN rm -rf .hack
+
+# go-modules downloads Crossplane's go modules. It's the base target of most Go
+# related target (go-build, etc).
+go-modules:
+  ARG NATIVEPLATFORM
+  FROM --platform=${NATIVEPLATFORM} golang:${GO_VERSION}
+  WORKDIR /crossplane
+  CACHE --id go-build --sharing shared /root/.cache/go-build
+  COPY go.mod go.sum ./
+  RUN go mod download
+  SAVE ARTIFACT go.mod AS LOCAL go.mod
+  SAVE ARTIFACT go.sum AS LOCAL go.sum
+
+# go-modules-tidy tidies and verifies go.mod and go.sum.
+go-modules-tidy:
+  FROM +go-modules
+  CACHE --id go-build --sharing shared /root/.cache/go-build
+  COPY --dir apis/ cmd/ internal/ pkg/ test/ .
+  RUN go mod tidy
+  RUN go mod verify
+  SAVE ARTIFACT go.mod AS LOCAL go.mod
+  SAVE ARTIFACT go.sum AS LOCAL go.sum
+
+# go-generate runs Go code generation.
+go-generate:
```

Comment:
Do we no longer need to do CRD patching of `DeploymentRuntimeConfig`? e.g., https://github.com/crossplane/crossplane/blob/master/Makefile#L87-L92

If so, can you help me explain how that is avoidable now? I came here looking for that specifically given what I had seen in https://github.com/crossplane/crossplane/pull/5734#issuecomment-2131300291

Reply:
It's still needed. It was added after I started working on this PR. I'll update.

Reply:
Just pushed a commit that implements it.